base/parse.jl

```diff
@@ -218,20 +218,22 @@ function parse(::Type{T}, s::AbstractString; base::Union{Nothing,Integer} = noth
 end
 
 ## string to float functions ##
-function parsefloat_preamble(s::AbstractString, base::Int, i::Int=1)
```

Comment:
You can just use `i::Int=firstindex(s)` if you want to keep this as an optional argument.

Reply:
i actually didn´t see a need for the optional argument to an internal function.

Using `startpos` and `endpos` may be useful to eg. strip currency symbols. But i think this is orthogonal to this PR.